debian/preinst

```diff
@@ -28,6 +28,8 @@ case "$1" in
                 rm -f $CACHE_DIR/$CACHE_FILE || true
             fi
         fi
+	# Remove old key on upgrade path
+        rm -f /etc/apt/trusted.gpg.d/ubuntu-esm-keyring.gpg
```

Comment:
Will this cause problems if the installation of a new version fails? Could we end up deleting `ubuntu-esm-keyring.gpg` but still be on the old version that requires it?

Reply:
per side conversation, I'll look at using dpkg-maintscript-helper's rm_conffile to handle this to avoid permanently deleting this file across failed package upgrade boundaries

Reply:
Ok, I looked over using dpk-maintscript cm_conffile and TLDR, I don't think we can use it because the old deb doesn't actually deliver this file in it's catalog. it is a side-effect of an enable-esm operation. 
As a result, the upgrade path won't see /etc/apt/trusted.gpg.d/ubuntu-esm-keyring.gpg as belonging to the upgraded deb package and so we can't rely on rm_conffile to remove it

Reply:
I think I'll leave this as is with an rm -f in preinst and we can chat about improvements for this next week with packaging folks.

Reply:
I think we're better leaving the file in place than risking breaking people's installations. Can we file an issue for the follow-up proper fix and then drop this `rm` for now?